### PR TITLE
[TwigComponent] Optimize TwigPreLexer

### PR DESCRIPTION
**Pull request by smnandre**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| Issues        | Fix #...
| License       | MIT

Refactor the following methods:
* `check`
* `consume`
* `consumeComponentName`
* `consumeUntil`
* `consumeWhitespace`

Remove the following methods:
* `addDefaultBlock`
* `consumeAttributeName`
* `doesStringEventuallyExist`

### CODE REVIEW

**Review by WebMamba — COMMENTED** [review]
Hey Simon! Maybe this is it out of scope of this PR, but what do you think to cache the PreLexer result ? Maybe we can do that in the ComponentLexer

**Comment by smnandre** [comment]
> Hey Simon! Maybe this is it out of scope of this PR, but what do you think to cache the PreLexer result ? Maybe we can do that in the ComponentLexer

I don't think we need to: 
- in prod all templates are cached 
- in dev, the lexer will be called only when the template changed (so we cannot use cache then)

**Review by kbond — APPROVED** [review]

File: src/TwigComponent/src/Twig/TwigPreLexer.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function preLexComponents(string $input): string
             return $input;
         }
 
-        $this->input = $input;
```
Comment (Kocal) [inline src/TwigComponent/src/Twig/TwigPreLexer.php:42]: I am surely wrong but, removing whitespaces won't increase the difficulty for the developper to know where it failed when we indicate the line number in an exception?

**Review by Kocal — APPROVED** [review]

File: src/TwigComponent/src/Twig/TwigPreLexer.php (hunk 1 above)
Comment (smnandre) [reply src/TwigComponent/src/Twig/TwigPreLexer.php:42]: I think i did this for these tests 

```php
$this->line += substr_count($content, "\n");
```

(and Twig Lexer start by doing this too, soo we need to keep the same line numbers)

Comment (smnandre) [reply src/TwigComponent/src/Twig/TwigPreLexer.php:42]: Sorry ii think i missunderstood at first. 

We are not removing any whitespace here (Twig Lexer will do it), we're just normalizing the new lines to simplify things, but there is no change as `\r\n` and `\r` are new line markers as `\n`

**Pull request merged**